docker-compose/docker-compose.yml

```
@@ -52,6 +52,19 @@ services:
       driver: fluentd
       options:
         fluentd-address: 127.0.0.1:24224
+    healthcheck:
+      test:
+        - "CMD"
+        - "bash"
+        - "-c"
+        - |
+          exec 3<>"/dev/tcp/localhost/1081"
+          echo -e "GET /liveness HTTP/1.1\r\nHost: localhost:1081\r\nConnection: close\r\n\r\n">&3 &
+          cat <&3 | grep Healthy || exit 1 && echo "$$?"
```

```suggestion
          if grep Healthy <&3 &>/dev/null ; then
            echo "$?"
          else
            exit 1
          fi
```

BTW, why do you have `"$$?"

---

Also, why do you do this echo?

---

I was just a really dirty way to get to emulate the `if` condition, (I had no idea I can "hardcode" bash code in the .yml in the way you suggested ^^). It's no longer necessary

---

Ok so you can just keep the grep only:
```suggestion
          grep Healthy <&3 &>/dev/null
```

And be sure that it is the last command

---

Docker compose expects either zero or one as output of your test if you feed it with other values, (in particular with a two) it might do unexpected stuff crazy

---

Tru, I forgot about that. So jsut add `|| exit 1` and nothing else